### ISSUE TITLE
Add app to Homebrew casks/MacPorts package managers to handle autoupdates

### ISSUE THREAD

**Issue by eugenesvk** [woelper/oculante#93]

**Comment by woelper** [comment]
This is an interesting topic. Would you be able to help out with this?
Are you aware oculante can auto-update through settings?

**Comment by eugenesvk** [comment]
I can help with Windows (see https://github.com/ScoopInstaller/Extras/issues/10003)

Don't know anything about Ports, and also unlikely can help with Brew since it doesn't support Catalina anymore, so might be a challenge to test, but might test it later, hopefully should be easier for binary installs...

Yes, I'm aware of auto-update, it's nice to have as well, but then it's not always the best thing for the first launch experience to be having to remember updating an app (especially when it's manual)

**Comment by eugenesvk** [comment]
Tried to add a cask (binary distributable) on Homebrew, but apparently casks on ARM **must** be signed and since your binary isn't it will not be accepted (doesn't pass the audit)
You might want to add your [own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap), this is the relevant cask formula that I created before I learned about the signing requirement
https://github.com/eugenesvk/homebrew-cask/blob/oculante/Casks/oculante.rb

Also, not sure if the most annoying part — updating the tap version/sha on new releases — can be automated

**Comment by eugenesvk** [comment]
Good news — I've managed to find a way for you to set up an automated workflow to update the homebrew cask on every new tag released on this main repo (there is a github action to open a pull request to update the version and another action to automerge it, so hopefully you won't have to do anything once it's set up)

You'd need to complete the following two steps:
- Copy this cask repo https://github.com/eugenesvk/homebrew-oculante and update the installation instructions with the new links (and then reference it in your workflow described below)
- Check out this [sample repo](https://github.com/eugenesvk/homebrew-bump) that would be the equivalent of your main repo https://github.com/woelper/oculante/, the ReadMe should have sufficiently described how to set up the autoupdates

(for example, if I push a tag in that `homebrew-bump` repo, then the `homebrew-oculante and` cask repo is fully autoupdated with the version taken from that tag, including automerging the PR)

**Comment by woelper** [comment]
Thank you, this sounds amazing! I'll get on with that as soon as I fixed some of the issues I am working on.

**Comment by Stoppedpuma** [comment]
@eugenesvk are you still interested in helping out with this? Woelper has since paid for the requirements to sign binaries on windows and mac platforms and we are basically just waiting on implementing a proper CLI test (brew docs recommend against using --help or similar if possible), figuring out a libheif bug on ARM macs (this would be especially awesome if you have one and would be willing to help us debug as Woelper can't reproduce this and I don't own an ARM mac), and publishing 0.9.3.

A list of applicable issues can be found at #664. It's a duplicate of this issue but I'll keep it open for now so you can decide on which issue you would prefer to use if you want to help out!

Additionally, we were a bit confused on the docs as it states that formulae can have a GUI if it has CLI functionality (we plan to extend a lot to CLI like filters and some other functions), but I at least am confused in the part that it seems to be only for CLI since some programs will have their GUI stripped to be a formulae. The reason it matters for if we can get it as formulae or not would be that it *should* fix any issues with libheif since formulae can build from source. Would you happen to know anything about this? A cask is fine if that's what we're left with but we do have to figure out a fix for libheif in that case which would be awesome as mentioned above if you would be willing to help test.

Thanks!

**Comment by eugenesvk** [comment]
I've updated my formula to reflect oculante's renamed release artifacts with intel/arm split so it should work for 0.9.2

Sorry, can't help you with debugging libheif.

Also don't know about such Brew details, where did you find it? in https://docs.brew.sh/Formula-Cookbook I only see 

> If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc

maybe you could have both with a clean separation?
  - a cask for pre-built GUI app and
  - a formula for the CLI that would also build libheif if needed?

and then set the required dependencies in the cask/formula (e.g., make the cask depend on the formula).

**Comment by Stoppedpuma** [comment]
> I've updated my formula to reflect oculante's renamed release artifacts with intel/arm split so it should work for the 0.9.2 release.

Thanks so much! I'll have to resume work on this at some point later as I'm a bit short on time at the moment but I plan to have this ready somewhat soon. The documentation I was talking about is at https://docs.brew.sh/Acceptable-Formulae under the "Stuff that builds a GUI by default (but doesn’t have to)" section.

> maybe you could have both with a clean separation?

I believe what me and woelper decided on trying was to attempt uploading as a formulae first, and if they didn't accept it to just upload it as a cask and try to figure out the libheif weirdness.

> Sorry, can't help you with debugging libheif.

No worries, I really meant just if you had an arm mac and could test that build. Thanks though!

Thanks again for helping out!

**Comment by eugenesvk** [comment]
To me the previous point is pretty clear:

> Stuff that builds an .app
Don’t make your formula build an .app (native macOS Application); we don’t want those things in Homebrew

Or did you mean to add another gui that's not an `.app`?

**Comment by Stoppedpuma** [comment]
I completely skipped over that, I think we'll just have to figure out the libheif issue and do a cask instead of a formulae.

**Comment by Stoppedpuma** [comment]
A small update on this, the libheif issue has been figured out. All we have left to do is signing the application in future releases. We should also be able to put Oculante on the App Store as well!

There's a few things to do before we do a minor release but it will hopefully be the first signed release so we can get the brew package up. This has been on the back burner for far too long.

Thanks again for your work on this! I'll co-author you on all the related PR's that are opened when we're ready!
